### DISCUSSION TITLE
Nested form data

### DISCUSSION THREAD

**Discussion by adriangb** [OAI/OpenAPI-Specification#2751] [Q&A]
In `multipart/form-data` requests each field has it's own `content-type`.
So (I think) you can have nested `multipart/form-data` fields, or at least `multipart/form-data` -> `multipart x-www-form-urlencoded`.
Does the spec support this sort of request body?

Thanks!

**Comment by handrews** [comment]
The `Content-Type` header has to go in the `contentType` field of the Encoding Object.  But I don't see any way you can easily describe that situation as there's nowhere to put a nested Media Type Object.  In OAS 3.1 you could _maybe_ do something very complicated with the JSON Schema `contentMediaType` keyword, but the chances of finding any tooling support is slim.  It feels like too niche of a use case to justify the complexity that would be involved in adding recursive nesting.

Since this feels like a simple yes/no answer and it's several years old, I'm going to close this, but feel free to comment if I have missed something and I can re-open it.  If you want to request this as a feature, you probably want to do so in the [OAS 4 "Moonwalk" discussions](https://github.com/OAI/sig-moonwalk/discussions) as it it too big of a change for a minor release.  But it does feel too complex for the benfit to me (my personal opinion only, not the opinion of the project!)

**Comment by handrews** [comment]
See also #3721 where this has also come up - idk if it will fit in 3.2 or Moonwalk, but if it can go in 3.2 it will be tracked on that issue.
